### PR TITLE
.create_basic_container added to LdpWriter

### PR DESCRIPTION
**Pull request by ndushay**
* .create_basic_container added to LdpWriter
* removed app/services/root_annotation_creator in favor of .create_basic_container

connected to #132

### CODE REVIEW

**Comment by ndushay** [comment]
@darrenleeweber  would like to get this merged ASAP so I can cut a new gem and update triannon on our servers (Fedora 4.1.1 is going in today)

**Comment by azaroth42** [comment]
Accepting the PR after talking through with @ndushay, on the grounds that it's blocking deployment of -stage by @eefahy. 

@darrenleeweber please read over the changes so you're familiar with them (no need to critique, just to sanity check and understand)

Thanks!

File: app/services/triannon/ldp_writer.rb [hunk 1]
```diff
@@ -61,6 +62,68 @@ def self.container_exist? path
...
+      slug.strip!
+      slug = slug[1..-1] if slug.start_with?('/')
+      if parent_path
+        parent_path.strip!
+        parent_path = parent_path[1..-1] if parent_path.start_with?('/')
+        parent_path.chop! if parent_path.end_with?('/')
+      end
+
+      full_path = (parent_path ? parent_path + '/' : "") + slug
+      full_url = base_url + '/' + full_path
```
Comment (darrenleeweber) [inline app/services/triannon/ldp_writer.rb]: Wondering whether all the path manipulation for starting and ending '/' might be easier with something like:
```ruby
full_path = [parent_path, slug].join('/')
path_array = full_path.split('/')
full_path = path_array.join('/')
```
Or maybe the `RDF::URI.new(parent_path, slug)` will take care of everything?

**Comment by darrenleeweber** [comment]
OK, I was on a conference call with Cornell and Harvard to discuss integration architectures for authority and catalog records.

File: app/services/triannon/ldp_writer.rb (hunk 1 above)
Comment (ndushay) [inline app/services/triannon/ldp_writer.rb]: I was having trouble with double slashes.  Agree that it's wordy.  But there is also a rule "working code wins."

**Pull request merged**